tooling/maven-plugin/src/main/java/org/apache/camel/quarkus/maven/UpdateExtensionDocPageMojo.java

```diff
@@ -316,6 +320,29 @@ static boolean detectNativeSsl(Path deploymentBasePath) {
         }
     }
 
+    static boolean detectAllowContextMapAll(CqCatalog catalog, String artifactId) {
+        final String allowContextMapAll = "allowContextMapAll";
+        Optional<ArtifactModel<?>> optional = catalog.filterModels(artifactId).findFirst();
```

Comment:
This may not work as expected if the first model is a LanguageModel or a Dataformat model. I think it would be safer to 
```suggestion
        return catalog.filterModels(artifactId)
                .filter(m -> m instanceof ComponentModel)
                .map(m -> (ComponentModel) m)
                .anyMatch(/* your options condition here */)
```

Reply:
Thanks. Latest commit implements this suggestion.